### PR TITLE
EDIFACT EDI Connector - Update the Reference

### PR DESCRIPTION
**Pull request by dhirsch1**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 1]
```diff
@@ -8,194 +8,391 @@
...
+| Unique message numbers | Boolean a|  Send unique Message Reference Numbers (UNH.1):
+
+* true
++
+Assigns unique transaction set numbers across all interchanges sent to the same partner, as determined by the interchange sender and recipient identification.
++
+* false
++
+Assigns message reference numbers sequentially within each interchange and reuses them in different interchanges. |  false |
++
```
Comment (sms-dchen) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]: I think this `+` is out of place

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 2]
```diff
@@ -8,194 +8,391 @@
...
 |===
 | Name | Type | Description | Default Value | Required
 | Configuration | String | The name of the configuration to use. | |x
-| Write Content | Object |  |  `#[payload]` |
-| Streaming Strategy | repeatable-in-memory-stream
-repeatable-file-store-stream
-non-repeatable-stream |  Configure to use repeatable streams and their behavior. |  |
-| Target Variable | String |  The name of a variable on which the operation's output is placed. |  |
-| Target Value | String |  An expression to evaluate against the operation's output and the outcome of that expression is stored in the target variable. |  `#[payload]` |
+| Write Content | Object | Output payload. |  `#[payload]` |
```
Comment (sms-dchen) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
| Write Content | Object | Input payload. |  `#[payload]` |
```
This is the input payload that the Write operation uses

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 3]
```diff
@@ -8,194 +8,391 @@
...
 
 [%header%autowidth.spread]
 |===
 | Name | Type | Description | Default Value | Required
-| Configuration | String | The name of the configuration to use. | |x
-| Write Content | Object |  |  `#[payload]` |
-| Target Variable | String |  The name of a variable on which the operation's output is placed. |  |
-| Target Value | String |  An expression to evaluate against the operation's output and the outcome of that expression is stored in the target variable. |  `#[payload]` |
+| Configuration | String | Name of the configuration to use. | |x
+| Write Content | Object | Output payload. |  `#[payload]` |
```
Comment (sms-dchen) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
| Batch Content | Binary | Input payload. |  `#[payload]` |
```
I noticed that it has the same value as the write operation, however, it has another name and it is a combination of the read and write operations. The input payload is an input stream (binary) and the output is also binary.

**Review by sms-dchen — DISMISSED** [review]
LGTM only minor changes

**Review by sms-dchen — DISMISSED** [review]

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 4]
```diff
@@ -8,194 +8,390 @@
...
+* false
++
+The connector uses the values and does not reject the messages. 
+
+In either case, the error is reported in CONTRL functional acknowledgments. |  true |
+| Enforce valid characters | Boolean a|  Enforce valid characters when using syntax levels UNOA and UNOB:
+
+* true
++
+The connector rejects received messages that contains characters that the syntax level does not allow.
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
The connector rejects received messages that contain characters that the syntax level does not allow.
```

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 5]
```diff
@@ -8,194 +8,390 @@
...
+{sp}+
+If set, this value affects both send and receive message handling.
+
+* On the receive side, the UNB interchange sender identification code qualifier must match this value. 
+* On the send side, the connector uses this value as the UNB interchange recipient identification code qualifier unless it is overridden in either the send configuration or the message parameters. |  |
+| Schema definitions | Array of String |  List of schema file system paths or class paths for the module to use. |  |x
+| Substitution character | String |  Substitution character used to replace invalid characters in string values when using syntax level `UNOA` or `UNOB`. The use of this substitution character, if set, is controlled by separate parser and writer flags. |  |
 | Separators usage | Enumeration, one of:
 
 USE_DEFAULTS
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
* USE_DEFAULTS
```

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 6]
```diff
@@ -8,194 +8,390 @@
...
 USE_SPECIFIED_FOR_WRITES
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
* USE_SPECIFIED_FOR_WRITES
```

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 7]
```diff
@@ -8,194 +8,390 @@
...
-USE_SPECIFIED_FOR_ALL |  Usage of specified separator and release characters:
-
-USE_DEFAULTS - Use defaults as defined by EDIFACT syntax version standards.
-USE_SPECIFIED_FOR_WRITES - Use standard default delimiters for reads, use specified delimiters as defaults for writes.
-USE_SPECIFIED_FOR_ALL - Use specified delimiters as defaults for both reads and writes. |  USE_SPECIFIED_FOR_ALL |
+USE_SPECIFIED_FOR_ALL a|  Defines how specified separator and release characters are used:
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
* USE_SPECIFIED_FOR_ALL a|  Defines how specified separator and release characters are used:
```

Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
* USE_SPECIFIED_FOR_ALL a|  Defines how specified separator and release characters are used:
```

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 8]
```diff
@@ -207,51 +403,58 @@ Merge individual EDIFACT messages created and accumulated over a period of time
...
+{sp}+
+This increases performance, but it also limits the number of concurrent requests your application can process, because it requires additional memory.
+* Configure a smaller buffer size to decrease memory load at the expense of response time.
+ |  |
+| Buffer Unit a| Enumeration, one of:
+
+* BYTE
+* KB
+* MB
+* GB | Unit for the Max In Memory Size field. |  |
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
* GB | Unit for the *Max In Memory Size* field. |  |
```

**Review by sathya0 — DISMISSED** [review]

File: edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 9]
```diff
@@ -8,194 +8,390 @@
...
+Always send a UNA service segment.
++
+* false
++
+Include a UNA in an interchange when the syntax separator characters don't match the defaults for the syntax version and character encoding in use. |  false |
+| Enforce valid characters | Boolean a| Enforce valid characters when using syntax levels UNOA and UNOB. 
+
+* true
++
+The connector rejects received messages that contains characters not allowed by the syntax level.
```
Comment (sathya0) [inline edifact-edi/2.7/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
The connector rejects received messages that contain characters not allowed by the syntax level.
```

**Review by sathya0 — APPROVED** [review]

**Pull request merged**